### PR TITLE
Cache access_token if enableTokenCaching true

### PR DESCRIPTION
**Pull request by valerii-cognite**
AuthProvider is now responsible for caching access_token in the localStorage so it can be retrieved and reused in authenticate.

### CODE REVIEW

**Comment by f1cognite** [comment]
@valerii-cognite Could you bump the SDK you are using? And set peerDependency to use version `>= 1.4.5` since then we can also get rid of `idToken`

**Comment by valerii-cognite** [comment]
@f1cognite done

File: src/ReactAuthProvider.tsx [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 
+export const ACCESS_TOKEN = 'REACT_AUTH_PROVIDER_ACCESS_TOKEN';
+
 export interface AuthProviderProps {
   project: string;
   redirectUrl: string;
   errorRedirectUrl: string;
   children: JSX.Element[] | JSX.Element;
   loadingComponent?: JSX.Element[] | JSX.Element;
+  enableTokenCaching: boolean | null;
```
Comment (f1cognite) [inline src/ReactAuthProvider.tsx]: Sure it shouldn't be `enableTokenCaching?: boolean`?

File: src/__tests__/__snapshots__/ReactAuthProvider.test.tsx.snap [hunk 2]
```diff
@@ -47,3 +47,17 @@ exports[`ReactAuthProvider update token on tokenCallback 2`] = `
   }
 />
 `;
+
+exports[`ReactAuthProvider use cached token when the prop is set 1`] = `
+<div
+  authResult={
+    Object {
+      "accessToken": "abc",
+      "idToken": "abc",
```
Comment (f1cognite) [inline src/__tests__/__snapshots__/ReactAuthProvider.test.tsx.snap]: We should remove `idToken` from the tests

File: src/ReactAuthProvider.tsx (hunk 1 above)
Comment (valerii-cognite) [reply src/ReactAuthProvider.tsx]: sure, til

File: src/__tests__/__snapshots__/ReactAuthProvider.test.tsx.snap (hunk 2 above)
Comment (valerii-cognite) [reply src/__tests__/__snapshots__/ReactAuthProvider.test.tsx.snap]: done

**Pull request merged**